pentaho-js/src/test/javascript/cdf/components/TableComponent-spec.js

```diff
@@ -196,5 +196,27 @@ define([
       dashboard.update(tableComponent);
     });
 
+    /**
+     * ## The Table Component # can update when server side is true
+     */
+    it("can update when server side is true", function(done) {
```

Comment:
I think in this case we can keep the functional test added for the TableComponent and also add the unit tests, that are **missing**, to cover the changes in BaseQuery and LegacyQuery.